engine/src/main/java/org/terasology/engine/core/modes/StateLoading.java

```diff
@@ -234,7 +234,7 @@ private void popStep() {
         current = null;
         if (!loadProcesses.isEmpty()) {
             current = loadProcesses.remove();
-            logger.debug(current.getMessage());
+            logger.debug("{}", current.getMessage());
```

Comment:
This looks weird 🙄 Is this necessary because of some quirk in the rules?

The only this might do is defer the call to `toString` on whatever type `getMessage()` returns. If that is already string, we don't gain anything from this...

Reply:
My understanding was that the parametrization is what avoids the call in case log level is not debug, so with
`logger.debug(current.getMessage())` the call to `getMessage()` is done even if log level is "info"
while with
`logger.debug("{}", current.getMessage())` the call to `getMessage()` is only done if log level is "debug" but not if it's "info"

Is my understanding incorrect?

Admittedly, in this particular case the call might not be particularly impactful.

Reply:
> Is my understanding incorrect?

I think it indeed is. The only thing that is defered is the final call to `toString` I think.

Maybe this gets clearer if we write this part of the code like this:

```java
// executed regardless of the log-level
Message message = current.getMessage(); 
// message.toString() is only called if DEBUG logs are enabled (happens inside)
logger.debug("{}", message);
```

The parameterization can only help as far as Java itself allows it. Here, we are passing the value of `current.getMessage()` to the method, that is, we evaluate that part already when calling it.

To make it a truly lazy evaluation we'd need a supplier, either for the whole message or for the inidividual parameters (see https://github.com/MovingBlocks/Terasology/pull/5173#discussion_r1392577738). I don't think the following code works, but that's how it would look like:

```java
logger.debug(() -> current.getMessage());
```

Here, we pass a function that, if evaluated, will make the lookup on how to get the message from `current`.